### PR TITLE
LineUp v4 testing demo

### PR DESCRIPTION
**Pull request by oltionchampari**
closes lineupjs/lineupjs#249

**prerequisites**: 
 * [x] branch is up-to-date with the branch to be merged with, i.e. develop
 * [x] build is successful
 * [x] code is cleaned up and formatted 
 * [ ] tested with Firefox 52, Firefox 57+, Chrome 64+, MS Edge 16+

### Summary
 Created an html demo to test all the available column types in LineUp v4.

### CODE REVIEW

File: demo/all_column_types.html [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        // Number
+        .column(LineUpJS.buildNumberColumn('numberMap').asMap().label('NumberMap').width(150))
+
+        // Date
+        .column(LineUpJS.buildDateColumn('dateMap').asMap().label('DateMap').width(150))
+
+        // Link
+        .column(LineUpJS.buildStringColumn('string').label('Link').pattern('https://duckduckgo.com/?q=${escapedValue}').width(150))
+        .column(LineUpJS.buildStringColumn('strings').label('Links').pattern('https://duckduckgo.com/?q=${escapedValue}').asArray().width(150))
+        .column(LineUpJS.buildStringColumn('stringMap').label('LinkMap').pattern('https://duckduckgo.com/?q=${escapedValue}').asMap().width(150))
```
Comment (oltionchampari) [inline demo/all_column_types.html]: @thinkh  I'm not sure if this is correct, or how exactly it is supposed to look.

Comment (thinkh) [reply demo/all_column_types.html]: I updated the code

**Comment by thinkh** [comment]
@sgratzl Could you please have a look at the demo and help us how to set the renderer for the _Links_ and _CategoricalMap_ columns correctly? To me it looks like there is no suitable renderer available.

grafik

**Comment by sgratzl** [comment]
for the categoricalmap you can use a regular "table" renderer. it seems like there is no special table for now. 

similar fo a `links` column, similar to a strings column there is no special renderer yet. There are mostly here for the sake of completeness.

**Comment by thinkh** [comment]
Ok, thanks for clarification, then I will leave a comment about that.

**Pull request merged**